### PR TITLE
feat: add hyper completion spec

### PR DESCRIPTION
**Pull request by Deep-Codes**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Feature : adds specs for `hyper`

### CODE REVIEW

**Review by fedeci — COMMENTED** [review]
If I specify a folder as arg does it open that folder in hyper?
e.g. `hyper node/`

**Comment by Deep-Codes** [comment]
> If I specify a folder as arg does it open that folder in hyper?
> e.g. `hyper node/`

It does adding `folder` as template

Screenshot 2021-08-23 at 8 18 47 PM

File: dev/hyper.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      name: ["list", "ls"],
+      description: "List installed plugins",
+    },
+    {
+      name: ["list-remote", "lsr", " ls-remote"],
+      description: "List plugins available on npm",
+    },
+    {
+      name: ["search", "s"],
+      description: "Search for plugins on npm",
```
Comment (QuiiBz) [inline dev/hyper.ts:36]: If think this option takes an argument, which is a string to search on npm?

File: dev/hyper.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    },
+    {
+      name: ["uninstall", "u", "rm", "remove"],
+      description: "Uninstall plugin",
```
Comment (QuiiBz) [inline dev/hyper.ts:40]: Same as above but for the name of the plugin

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Awesome! There are some arguments that are missing thought

File: dev/hyper.ts (hunk 2 above)
Comment (fedeci) [reply dev/hyper.ts:40]: It can actually be autogenerated via the `list` subcommand.

Comment (QuiiBz) [reply dev/hyper.ts:40]: Yes! But adding generators isn't mandatory when adding a completely new spec. We can add them later!

File: dev/hyper.ts (hunk 1 above)
Comment (Deep-Codes) [reply dev/hyper.ts:36]: i don't think so 

```
  Usage: hyper [options] [command]
  
  Commands:
    <default>                    Launch Hyper
    docs, d, h, home             Open the npm page of a plugin
    help                         Display help
    install, i                   Install a plugin
    list, ls                     List installed plugins
    list-remote, lsr, ls-remote  List plugins available on npm
    search, s                    Search for plugins on npm
    uninstall, u, rm, remove     Uninstall a plugin
    version                      Show the version of hyper
  
  Options:
    -h, --help     Output usage information
    -v, --verbose  Verbose mode (disabled by default)
  
```

Comment (QuiiBz) [reply dev/hyper.ts:36]: Could you try running `hyper search help`?

Comment (Deep-Codes) [reply dev/hyper.ts:36]: this is the usage 

`> hyper search` doesn't take an argument 🤔 

Screenshot 2021-08-23 at 9 11 34 PM

Comment (Deep-Codes) [reply dev/hyper.ts:36]: Yes it is

Comment (QuiiBz) [reply dev/hyper.ts:36]: Just installed and tried it, and looks like I can add search arguments (`material` here). Maybe we don't have the same version of the cli?
Screenshot 2021-08-23 at 17 46 13

Comment (Deep-Codes) [reply dev/hyper.ts:36]: yes, it does search according to the argument.

Should i add, args to the search subcommand?

Screenshot 2021-08-23 at 9 21 33 PM

Comment (QuiiBz) [reply dev/hyper.ts:36]: Yes you can add it, along with the [isOptional](https://fig.io/docs/reference/arg#isoptional) option!

File: dev/hyper.ts (hunk 2 above)
Comment (Deep-Codes) [reply dev/hyper.ts:40]: Can you explain What should I do here?

Comment (QuiiBz) [reply dev/hyper.ts:40]: We were talking about [generators](https://fig.io/docs/reference/generator), which is an amazing feature of Fig. They allow you to dynamically provide suggestions, using a shell command. In this case, we could make a generator for `uninstall`, which would run the `hyper ls` command in the background, parse the result, and output it as suggestions.

You don't need to make generators right now, it's just a cool feature that can be added in the future. Feel free to ask if you need more help on this!

File: dev/hyper.ts (hunk 1 above)
Comment (Deep-Codes) [reply dev/hyper.ts:36]: Added ✅

File: dev/hyper.ts (hunk 2 above)
Comment (Deep-Codes) [reply dev/hyper.ts:45]: Hey got it running 🚀

Screenshot 2021-08-23 at 10 57 30 PM

**Comment by Deep-Codes** [comment]
Seems like a new lint rule is failing CI 🙂

**Comment by fedeci** [comment]
You can rebase on master, run `npm run lint:fix` and we are good to go!

**Comment by Deep-Codes** [comment]
@fedeci i think this is done 🙂

**Comment by QuiiBz** [comment]
Could you rebase your PR to `master`? There are a lot of past commits above.

**Review by QuiiBz — APPROVED** [review]
Just removed the past commits, thanks a lot for this PR!

**Pull request merged**